raft/server.hh

```diff
@@ -43,8 +45,16 @@ public:
         bool enable_forwarding = true;
 
         // Max size of a single command, add_entry with a bigger command will throw command_is_too_big_error.
-        // The value of zero means no limit.
-        size_t max_command_size = 0;
+        // The following condition must be satisfied:
+        // max_command_memory_usage() <= max_log_size - max_snapshot_trailing_bytes
+        // A snapshot will be triggered if memory_usage reaches _config.max_log_size - _config.max_command_memory_usage(),
+        // therefore, to avoid too frequent snapshots, it makes sense to make
+        // max_log_size several times larger than max_command_size.
+        size_t max_command_size = 100 * 1024;
+
+        size_t max_command_memory_usage() const {
+            return max_command_size + sizeof(log_entry);
```

Comment:
For the purposes of the patch, i.e. limiting the command size and providing backpressure, you don't need to account for log entry size at all. Removing this accounting would reduce the number of variables to keep track of and simplify calculations.

Reply:
> For the purposes of the patch, i.e. limiting the command size and providing backpressure, you don't need to account for log entry size at all. 

The purpose of the patch is to provide backpressure in case the log consumes too much memory, originally dtest was crashing for this reason, by OOM. If command size is small (a few bytes), it may be much smaller than `sizeof(log_entry) == 136`, the memory usage calculated only by command size will be much smaller than actual memory usage (several times) and we will get the same OOM.

Reply:
dtest was crashing because the entry couldn't fit the commit log, not because the in-memory log was taking too much memory.

Reply:
There were tho different failures in two different tests. One for the first reason and another for the second.

Reply:
there were two tests
imits_test.py::TestLimits::test_max_columns_and_query_parameters
limits_test.py::TestLimits::test_max_cells
one was failing because because of the single command being too big, another one - because the log took too much space. At least, thats how I understood gleb.

Reply:
> there were two tests imits_test.py::TestLimits::test_max_columns_and_query_parameters limits_test.py::TestLimits::test_max_cells one was failing because because of the single command being too big, another one - because the log took too much space. At least, thats how I understood gleb.

Right.